backend/sozisel/lib/sozisel_web/schema/resolvers/session_resource_resolvers.ex

```diff
@@ -0,0 +1,165 @@
+          {:error, :enoent} ->
+            {:ok, %{}}
+
+          error ->
+            error
+        end
+      end)
+      |> Repo.transaction()
+      |> case do
+        {:ok, %{resource: session_resource}} ->
+          {:ok, session_resource}
+
+        {:error, operation, value, _others} ->
+          Logger.error(
+            "Failed to delete session resource: #{inspect(operation)}, #{inspect(value)}"
+          )
+
+          {:error, "failed to delete session resource"}
+      end
+    else
+      nil ->
+        {:error, "session resource does not exist"}
+    end
+  end
+
+  def attach_resource_to_session(
+        _parent,
+        %{input: %{resource_id: resource_id, session_id: session_id, is_public: is_public}},
+        _ctx
+      ) do
+    with %SessionResource{} = resource <- Repo.get_by(SessionResource, id: resource_id),
+         %Session{} = session <- Repo.get_by(Session, id: session_id),
+         {:ok, %SessionResourceLink{} = session_resource_link} =
+           SessionResourceLinks.create_session_resource_link(%{
+             is_public: is_public,
+             session_id: session.id,
+             session_resource_id: resource.id
+           }) do
+      {:ok, %{id: session_resource_link.id, is_public: is_public, session_resource: resource}}
+    else
+      nil ->
+        Logger.error("resource or session do not exist")
+        {:error, nil}
+    end
+  end
+
+  def detach_resource_session_link(
+        _parent,
+        %{id: id},
+        _ctx
+      ) do
+    with %SessionResourceLink{} = session_resource_link <-
+           Repo.get(SessionResourceLink, id)
+           |> Repo.preload(:session_resource) do
+      {:ok, %SessionResourceLink{}} =
+        SessionResourceLinks.delete_session_resource_link(session_resource_link)
+
+      {:ok, session_resource_link}
+    else
+      nil ->
+        {:error, "session resource link does not exist"}
+    end
+  end
+
+  def change_access_resource(_parent, %{id: id, is_public: is_public}, _ctx) do
+    with %SessionResourceLink{} = session_resource_link <-
+           Repo.get(SessionResourceLink, id)
+           |> Repo.preload(:session_resource) do
+      {:ok, %SessionResourceLink{} = session_resource_link} =
+        SessionResourceLinks.update_session_resource_link(session_resource_link, %{
+          is_public: is_public
+        })
+
+      {:ok, session_resource_link}
+    else
+      nil ->
+        {:error, nil}
+    end
+  end
+
+  def get_presenter_session_resources(_parent, %{id: id}, _ctx) do
+    session_resource_links =
+      SessionResourceLinks.list_session_resource_links(%{session_id: id})
+      |> Repo.preload(:session_resource)
+
+    {:ok, session_resource_links}
+  end
+
+  def get_participant_session_resources(_parent, %{id: id}, _ctx) do
+    session_resource_links =
+      %{session_id: id, is_public: true}
+      |> SessionResourceLinks.list_session_resource_links()
+      |> Repo.preload(:session_resource)
```

Comment:
You don't have to preload those session resources as dataloader should do that, not you